ruby/tests/api-test.cc

```diff
@@ -0,0 +1,132 @@
+        REQUIRE_FALSE(ruby->is_string(1));
+    }
+
+    SECTION("can correctly identify symbols") {
+        REQUIRE(ruby->is_symbol(ruby->to_symbol("mysymbol")));
+        REQUIRE_FALSE(ruby->is_symbol(ruby->false_value()));
+    }
+
+    SECTION("can correctly identify numbers") {
+        REQUIRE(ruby->is_float(ruby->eval("1.5")));
+        REQUIRE_FALSE(ruby->is_float(ruby->utf8_value("foo")));
+
+        REQUIRE(ruby->is_fixednum(ruby->eval("2")));
+        REQUIRE_FALSE(ruby->is_fixednum(ruby->eval("1.5")));
+    }
+
+    SECTION("can correctly identify hashes") {
+        REQUIRE(ruby->is_hash(ruby->eval("{ 'red' => 2 }")));
+        REQUIRE_FALSE(ruby->is_hash(ruby->utf8_value("foo")));
+    }
+
+    SECTION("can correctly identify type") {
+        REQUIRE(ruby->is_a(ruby->eval("1"), ruby->eval("Integer")));
+        REQUIRE_FALSE(ruby->is_a(ruby->eval("'1'"), ruby->eval("Integer")));
+    }
+
+    SECTION("can correctly identify arrays") {
+        REQUIRE(ruby->is_array(ruby->eval("[1, 2, 3]")));
+        REQUIRE_FALSE(ruby->is_array(ruby->false_value()));
+    }
+}
+
+TEST_CASE("api::equals", "[ruby-api]") {
+    auto ruby = api::instance();
+    REQUIRE(ruby);
+    ruby->initialize();
+    REQUIRE(ruby->initialized());
+    SECTION("can correctly test boolean values for equality") {
+        REQUIRE(ruby->equals(ruby->true_value(), ruby->true_value()));
+        REQUIRE_FALSE(ruby->equals(ruby->true_value(), ruby->false_value()));
+    }
+
+    SECTION("can correctly test strings for equality") {
+        REQUIRE(ruby->equals(ruby->utf8_value("foo"), ruby->utf8_value("foo")));
+        REQUIRE_FALSE(ruby->equals(ruby->utf8_value("foo"), ruby->utf8_value("bar")));
+    }
+
+    SECTION("can correctly test numbers for equality") {
+        REQUIRE(ruby->equals(1, 1));
+        REQUIRE_FALSE(ruby->equals(1, 3));
+        REQUIRE(ruby->equals(ruby->eval("1.5"), ruby->eval("1.5")));
+        REQUIRE_FALSE(ruby->equals(ruby->eval("1.5"), ruby->eval("1")));
+    }
+
+    SECTION("can correctly test Ruby hashes for equality") {
+        REQUIRE(ruby->equals(ruby->eval("{ 'red' => 'blue' }"), ruby->eval("{ 'red' => 'blue' }")));
+        REQUIRE_FALSE(ruby->equals(ruby->eval("{ 'red' => 'blue' }"), ruby->eval("{ 'red' => 'green' }")));
+    }
+
+    SECTION("can correctly test symbols for equality") {
+        REQUIRE(ruby->equals(ruby->to_symbol("mysymbol"), ruby->eval(":mysymbol")));
+        REQUIRE_FALSE(ruby->equals(ruby->to_symbol("mysymbol"), ruby->to_symbol("notmysymbol")));
+    }
+}
+
+TEST_CASE("api::case_equals", "[ruby-api]") {
+    auto ruby = api::instance();
+    REQUIRE(ruby);
+    ruby->initialize();
+    REQUIRE(ruby->initialized());
+
+    SECTION("can detect class membership") {
+        REQUIRE(ruby->case_equals(ruby->eval("Integer"), ruby->eval("1")));
+        REQUIRE_FALSE(ruby->case_equals(ruby->eval("String"), ruby->eval("4")));
+    }
+}
+
+VALUE test_func(VALUE self) {
```

Comment:
Keeping function declarations local to where they're used works for me, until we have a reason not to. So this seems fine.